pkg/controllers/management/rkeworkerupgrader/upgrade.go

```diff
@@ -474,6 +480,14 @@ func (uh *upgradeHandler) retryClusterUpdate(name string) error {
 
 }
 
+func keys(nodes []*v3.Node) []string {
+	keys := []string{}
```

Comment:
Create this with the exact length you need so the underlying array won't need to be grown, this will mainly affect large clusters.

Reply:
`keys := make([]string, len(nodes))`